tests/components/starlink/test_config_flow.py

```diff
@@ -0,0 +1,52 @@
+"""Test the Starlink config flow."""
+from unittest.mock import patch
+
+from starlink_grpc import GrpcError
+
+from homeassistant import data_entry_flow
+from homeassistant.components.starlink.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+
+from .patchers import SETUP_ENTRY_PATCHER
+
+
+async def test_flow_user_fails_no_dishy(hass: HomeAssistant) -> None:
+    """Test user initialized flow when Starlink is available."""
+    user_input = {CONF_IP_ADDRESS: "192.168.100.1:9200"}
+
+    with patch(
+        "starlink_grpc.get_id", side_effect=GrpcError("Some exception")
+    ), SETUP_ENTRY_PATCHER:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+        )
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input=user_input,
+        )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["errors"]
+
+
+async def test_flow_user_success(hass: HomeAssistant) -> None:
```

Comment:
I'm not sure what's wrong here, but the second test always fails.
For example, if I run the test as-is now, `test_flow_user_success` will fail. If I swap the order of the tests, then it will be `test_flow_user_fails_no_dishy` will be the one failing.
The failure is from `assert result["type"]`, where the type of the second test matches the first test (thus the assertion fails)
If anyone has any ideas, please let me know

Reply:
Fixed, I ended up patching the `get_device_id` in the config flow class